workflows/references/main.nf

```diff
@@ -72,81 +79,76 @@ workflow REFERENCES {
         versions = versions.mix(BOWTIE2_BUILD.out.versions)
     }
 
-    if (tools && tools.split(',').contains('bwamem1')) {
-        BWAMEM1_INDEX(input.fasta)
-
-        bwamem1 = BWAMEM1_INDEX.out.index
-        versions = versions.mix(BWAMEM1_INDEX.out.versions)
-    }
-
-    if (tools && tools.split(',').contains('bwamem2')) {
-        BWAMEM2_INDEX(input.fasta)
-
-        bwamem2 = BWAMEM2_INDEX.out.index
-        versions = versions.mix(BWAMEM2_INDEX.out.versions)
-    }
-
-    if (tools && tools.split(',').contains('dragmap')) {
-        DRAGMAP_HASHTABLE(input.fasta)
-
-        dragmap = DRAGMAP_HASHTABLE.out.hashmap
-        versions = versions.mix(DRAGMAP_HASHTABLE.out.versions)
-    }
-
-    if (tools && tools.split(',').contains('createsequencedictionary')) {
-        GATK4_CREATESEQUENCEDICTIONARY(input.fasta)
-
-        dict = GATK4_CREATESEQUENCEDICTIONARY.out.dict
-        versions = versions.mix(GATK4_CREATESEQUENCEDICTIONARY.out.versions)
-    }
+    // the whole map -> null should be removed once I managed to make it work properly
+    BWAMEM1_INDEX(
```

Comment:
This is a really nice way to skip the repeated if statement.
On the other hand this will pollute the output nextflow log with empty processes...
Is it okay for you ?

Reply:
that's fine for now, I'll deal with this later if it's really an issue, but for now, I just want a code that is cleaner to read

Reply:
In phaseimpute we also have this issue with branching event, but nothing can be done.